resources/providers/awslib/ec2/provider.go

```diff
@@ -38,44 +38,50 @@ type Client interface {
 	DescribeSecurityGroups(ctx context.Context, params *ec2.DescribeSecurityGroupsInput, optFns ...func(*ec2.Options)) (*ec2.DescribeSecurityGroupsOutput, error)
 }
 
-func (p *Provider) DescribeNeworkAcl(ctx context.Context) ([]awslib.AwsResource, error) {
-	allAcls := []types.NetworkAcl{}
-	input := ec2.DescribeNetworkAclsInput{}
-	for {
-		output, err := p.client.DescribeNetworkAcls(ctx, &input)
-		if err != nil {
-			return nil, err
-		}
-		allAcls = append(allAcls, output.NetworkAcls...)
-		if output.NextToken == nil {
-			break
+func (p *Provider) DescribeNetworkAcl(ctx context.Context) ([]awslib.AwsResource, error) {
+	var allAcls []types.NetworkAcl
+	for _, client := range p.Clients {
```

Comment:
This function involves too much logic and which causes confusion.
The old implementation wasn't "region aware" which is good.
The cross-region logic ideally should not be a part of that function, instead, we should implement it in a different layer above that function.